src/pages/settings/Profile/Contacts/NewContactMethodPage.js

```diff
@@ -52,99 +52,84 @@ const defaultProps = {
     loginList: {},
 };
 
-class NewContactMethodPage extends Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            login: '',
-            password: '',
-        };
-        this.onLoginChange = this.onLoginChange.bind(this);
-        this.validateForm = this.validateForm.bind(this);
-        this.submitForm = this.submitForm.bind(this);
-    }
-
-    onLoginChange(login) {
-        this.setState({login});
-    }
+function NewContactMethodPage(props) {
+    const [login, setLogin] = useState('');
+    const [password, setPassword] = useState('');
+    const loginInputRef = useRef(null);
 
     /**
     * Determine whether the form is valid
     *
     * @returns {Boolean}
     */
-    validateForm() {
-        const login = this.state.login.trim();
-        const phoneLogin = LoginUtils.getPhoneNumberWithoutSpecialChars(login);
+    const validateForm = useCallback(() => {
+        const phoneLogin = LoginUtils.getPhoneNumberWithoutSpecialChars(login.trim());
 
-        return (Permissions.canUsePasswordlessLogins(this.props.betas) || this.state.password)
+        return (Permissions.canUsePasswordlessLogins(props.betas) || this.state.password)
             && (Str.isValidEmail(login) || Str.isValidPhone(phoneLogin));
-    }
+    }, [login, password, props.betas]);
 
-    submitForm() {
+    const submitForm = useCallback(() => {
```

Comment:
It probably isn't strictly required here per the react docs https://react.dev/reference/react/useCallback#usage it suggests you probably only need it for a performance optimization.

But I chatted with @marcaaron a bit 1:1 about it, sounds like there was a discussion somewhere (not sure where probably open-source but I haven't looked for it yet) where we landed on preferring to just always use `useCallback` 

> the resolution was that we should just always use it since the performance downsides of missing when we should use it vs not can be pretty bad.
> ⁦Or said another way - we'll probably mess stuff up more by not using it then we will by using it

Reply:
That makes sense, thanks

Reply:
hmm interesting thread that seems to be saying the opposite of what we decided here https://expensify.slack.com/archives/C01GTK53T8Q/p1680096510744619

just wanted to share so others didn't keep thinking we should prefer `useCallback` by default.